client/app/recordings/new/route.js

```diff
@@ -7,11 +7,62 @@ export default Route.extend({
     this.container = options.container;
   },
 
-  render() {
-    this.view = new View({
-      model: new Model(),
-      editing: true
-    });
-    this.container.show(this.view);
-  }
+  render(params) {
+    var that = this;
+    function renderPage(_model){
+      that.view = new View({
+        model: _model,
+      });
+      that.container.show(that.view);
+    }
+    function retrieveListAndRender(_model){
+      var listxhr = new XMLHttpRequest();
+      listxhr.onreadystatechange = function(){
+        var list = JSON.parse(listxhr.responseText)["data"]
+        _model.attributes["organizationList"] = list;
+        renderPage(_model);
+      }
+      listxhr.open("GET", "http://localhost:6543/organizations");
+      listxhr.send();
+    }
+    var _model = new Model();
+    var hostName;
+    if(params){
+      params = params.split("&").map(function(pair){
+        return pair.split("=");
+      });
+      for(var i = 0; i < params.length; i++){
+        switch(params[i][0]){
+          case "title":
+            _model.attributes["title"] = params[i][1];
+            break;
+          case "url":
+            _model.attributes["url"] = params[i][1];
+            hostName = new URL(params[i][1]).host;
+            break;
+        }
+      }
+    } else {
+      retrieveListAndRender(_model);
+    }
+
+
+    if(hostName){
+      var xhr = new XMLHttpRequest();
+      xhr.open("GET", "http://localhost:6543/organizations?url=" + hostName, false);
```

Comment:
This should be collection logic, much like the above (it should be same code, really).